core/templates/dev/head/pages/exploration_editor/simple_editor_tab/simple_editor_sidebar_directive.html

```diff
@@ -12,13 +16,16 @@
 
     <li class="sidebar-item" ng-repeat="question in questionList.getQuestions() track by $index">
       <div id="<[ID_PREFIX + question.getId()]>" class="sidebar-item-header">
-        <i ng-click="question.toggleCollapse()" class="collapse-icon material-icons">
+        <i ng-click="question.toggleCollapse()" class="collapse-icon material-icons" ng-hide="sidebarModeService.isSidebarInEditMode()">
           <[isCollapsed(question) ? '&#xE5C7;' : '&#xE5C5;']>
         </i>
+        <i ng-click="deleteQuestion(question)" class="delete-icon material-icons" ng-hide="sidebarModeService.isSidebarInReadonlyMode()">
+          <['&#xE872;']>
+        </i>
         <strong ng-click="scrollToQuestion(question)">Question <[$index + 1]></strong>
       </div>
 
-      <ul ng-show="question.isCollapsed()">
+      <ul ng-show="question.isCollapsed() && sidebarModeService.isSidebarInReadonlyMode()">
```

Comment:
Actually, either is fine. There is one important difference between ng-if and ng-show though:

- ng-if actually removes the whole thing from the DOM if its condition is false.
- ng-show just sets its visibility to hidden.

With ng-if, you can also run into some issues when accessing parent scopes, since an ng-if directive creates a new isolate scope. So databinding can become wonky if you don't know what you're doing.